src/mapgen_flat.cpp

```diff
@@ -453,6 +453,8 @@ MgStoneType MapgenFlat::generateBiomes(float *heat_map, float *humidity_map)
 					stone_type = DESERT_STONE;
 				else if (biome->c_stone == c_sandstone)
 					stone_type = SANDSTONE;
+			} else {
+				continue;
```

Comment:
The conditionals below can use `biome` uninitialized if it isn't initialized in this block.

I was going to just invert the condition and deindent the block, but the condition was rather complex and I wasn't entirely confident that I'd get it right.